cypress/integration/common/ping.js

```diff
@@ -0,0 +1,17 @@
+import { Given } from "cypress-cucumber-preprocessor/steps";
+
+Given('I visit the ping page', (path) => {
+    // note that cy.visit will follow redirects, and require the status code to be 2xx after that
+    cy.visit('/ping');
+})
+
+Given('I visit the ping JSON endpoint', (path) => {
+    cy.request('/ping/json').then((response) => {
```

Comment:
Just wondering if the `then` bit could be split into a then step in the feature, something along the lines of  `Then I get a valid  Response` or similar.

Reply:
+1  That's a good point . The bit that checks it, should be in a Then step.  Given I visit the Ping JSON endpoint,  Then I get back valid json .   Then. "Then I get back valid json" step could then be reused in many places.

Reply:
this might be useful as we could split the tests up... https://example.cypress.io/commands/network-requests

Reply:
(see 3rd example about shared test context)

Reply:
I added a commit to make this more generic so it could be used for different endpoints. I also added a generic step for testing object properties. This can be used specifically for JSON, but is also general enough to work with other types of object.